vendor/bat-native-ads/src/bat/ads/internal/bundle/bundle.cc

```diff
@@ -203,6 +224,88 @@ std::unique_ptr<BundleState> Bundle::GenerateFromCatalog(
         }
       }
 
+      // New tab page ad creatives
+      for (const auto& creative : creative_set.creative_new_tab_page_ads) {
+        if (!DoesOsSupportCreativeSet(creative_set)) {
+          const std::string platform_name =
+              PlatformHelper::GetInstance()->GetPlatformName();
+
+          BLOG(1, "Creative set id " << creative_set.creative_set_id
+              << " does not support " << platform_name);
+
+          continue;
+        }
+
+        CreativeNewTabPageAdInfo info;
+        info.creative_instance_id = creative.creative_instance_id;
+        info.creative_set_id = creative_set.creative_set_id;
+        info.campaign_id = campaign.campaign_id;
+
+        base::Time start_at_time;
+        if (base::Time::FromUTCString(campaign.start_at.c_str(),
+            &start_at_time)) {
+          info.start_at_timestamp =
+              static_cast<int64_t>(start_at_time.ToDoubleT());
+        } else {
+          info.start_at_timestamp = std::numeric_limits<int64_t>::min();
+
+          BLOG(1, "Creative set id " << creative_set.creative_set_id
+              << " has an invalid startAt timestamp");
+        }
+
+        base::Time end_at_time;
+        if (base::Time::FromUTCString(campaign.end_at.c_str(),
+            &end_at_time)) {
+          info.end_at_timestamp =
+              static_cast<int64_t>(end_at_time.ToDoubleT());
+        } else {
+          info.end_at_timestamp = std::numeric_limits<int64_t>::max();
+
+          BLOG(1, "Creative set id " << creative_set.creative_set_id
+              << " has an invalid endAt timestamp");
+        }
+
+        info.daily_cap = campaign.daily_cap;
+        info.advertiser_id = campaign.advertiser_id;
+        info.priority = campaign.priority;
+        info.ptr = campaign.ptr;
+        info.conversion =
+            creative_set.ad_conversions.size() != 0 ? true : false;
+        info.per_day = creative_set.per_day;
+        info.total_max = creative_set.total_max;
+        info.geo_targets = geo_targets;
+        info.company_name = creative.payload.company_name;
+        info.alt = creative.payload.alt;
+        info.target_url = creative.payload.target_url;
+
+        // Segments
+        for (const auto& segment : creative_set.segments) {
+          auto segment_name = base::ToLowerASCII(segment.name);
+
+          std::vector<std::string> segment_name_hierarchy =
```

Comment:
Nit: I believe this allocates memory each this in this loop and the parent loop. I think if we just instantiate this once in above the parent loop, we will iterate faster.

Reply:
Discussed and not an issue